mcp/src/main/java/io/modelcontextprotocol/spec/McpServerSession.java

```diff
@@ -138,6 +149,13 @@ public Mono<Void> sendNotification(String method, Map<String, Object> params) {
 		return this.transport.sendMessage(jsonrpcNotification);
 	}
 
+	public Mono<Void> sendNotification(String method, Object params) {
```

Comment:
Why do we need to add another public method to the `McpServerSession`? We need to maintain consistency here and make sure the surface is as minimal as we can.

Reply:
because we don't have ObjectMapper outside the session and can't convert the object to map as expected by the schema.  Perhaps we need to make this as the public apis and remove the one that takes explicit map.

Reply:
Because the McpAsyncServerExchange doesn't have access to transport and can not convert the `LoggingMessageNotification` input into  `Map<String, Object> params`. 
I guess we need to relax the sendNotification signature to accept Object params instead of the Map. but this is a breaking change.  I will add  a comment on the method.

Reply:
Just to keep the record of where this ended up - #137 addressed an earlier mistake that triggered this problematic situation and we are now in a cleaner state :heart: